packages/frontend/app/lib/apollo.server.ts

```diff
@@ -26,50 +23,56 @@ BigInt.prototype.toJSON = function (this: bigint) {
   return this.toString()
 }
 
-const authLink = setContext((request, { headers }) => {
-  if (!process.env.SIGNATURE_SECRET || !process.env.SIGNATURE_VERSION)
-    return { headers }
-  const timestamp = Math.round(new Date().getTime() / 1000)
-  const version = process.env.SIGNATURE_VERSION
+// TODO: rm env vars? SIGNATURE_VERSION, SIGNATURE_SECRET
+async function createAuthLink(request: Request) {
+  return setContext(async (gqlRequest, { headers }) => {
+    if (!process.env.SIGNATURE_SECRET || !process.env.SIGNATURE_VERSION)
+      return { headers }
+    const timestamp = Math.round(new Date().getTime() / 1000)
+    const version = 1 // process.env.SIGNATURE_VERSION
+    const session = await getSession(request.headers.get('cookie'))
+    const apiSecret = session.get('apiSecret')
 
-  const { query, variables, operationName } = request
-  const formattedRequest = {
-    variables,
-    operationName,
-    query: print(query)
-  }
+    if (!apiSecret) {
+      // TODO: is this the correct behavior? same things as if sig secret and version are not set
+      return { headers }
+    }
 
-  const payload = `${timestamp}.${canonicalize(formattedRequest)}`
-  const hmac = createHmac('sha256', process.env.SIGNATURE_SECRET)
-  hmac.update(payload)
-  const digest = hmac.digest('hex')
+    const { query, variables, operationName } = gqlRequest
+    const formattedRequest = {
+      variables,
+      operationName,
+      query: print(query)
+    }
 
-  return {
-    headers: {
-      ...headers,
-      signature: `t=${timestamp}, v${version}=${digest}`
+    const payload = `${timestamp}.${canonicalize(formattedRequest)}`
+    const hmac = createHmac('sha256', process.env.SIGNATURE_SECRET)
+    hmac.update(payload)
+    const digest = hmac.digest('hex')
+
+    const link = {
+      headers: {
+        ...headers,
+        signature: `t=${timestamp}, v${version}=${digest}`
+      }
     }
-  }
-})
+
+    const tenantId = session.get('tenantId')
+    if (tenantId) {
+      link.headers['tenant-id'] = tenantId
+    }
+
+    return link
+  })
+}
 
 const httpLink = createHttpLink({
   uri: process.env.GRAPHQL_URL
 })
 
-if (!global.__apolloClient) {
-  global.__apolloClient = new ApolloClient({
+export async function getApolloClient(request: Request) {
+  return new ApolloClient({
     cache: new InMemoryCache({}),
-    link: ApolloLink.from([authLink, httpLink]),
-    defaultOptions: {
-      query: {
-        fetchPolicy: 'no-cache'
-      },
-      mutate: {
-        fetchPolicy: 'no-cache'
-      }
-    }
```

Comment:
no longer needed since clients arent shared between requests and we arent worried about accidentally sharing cache